main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+    "encoding/csv"
+    "github.com/gin-gonic/gin"
+    "io"
+    "log"
+    "os"
+    "strconv"
+)
+
+const CSV_ERROR_MESSAGE = "There was an error when attempting to read the csv file. Please try again later."
+const FILE_PATH = "covid.csv"
+
+type Country struct {
+    Name   string  `json:"country_name"`
+    Cases  int     `json:"cases"`
+    Deaths int     `json:"deaths"`
+    Ratio  float32 `json:"death_ratio"`
+}
+
+func csvError(context *gin.Context, err error) {
+    log.Println(err)
+    context.JSON(500, gin.H{
+        "message": CSV_ERROR_MESSAGE,
+    })
+}
+
+func covidHandler(context *gin.Context) {
+    file, err := os.Open(FILE_PATH)
+    log.Println("%T", err)
```

Comment:
As `err` is always an `error` variable, `%T` will always print `error`.

`%T` prints the kind of the variable

to print the error object using `println`:
```suggestion
    log.Println(err)
```
to print the string representation of the Error using `println`:
```suggestion
    log.Println (err.Error())
```
to better show the error message by using `printf`:
```suggestion
    log.Printf("Something failed when opening the file: %v", err)
```
or this one:
```suggestion
    log.Printf("Something failed when opening the file: %s", err.Error())
```

Reply:
I was figuring out what the type of the `err` variable was, forgot to remove that line, sorry about that.